### ISSUE TITLE
limiting number of CPUs with intra_op_parallelism_threads (tensorflow backend)

### ISSUE THREAD

**Issue by wdobbels** [keras-team/keras#9936]
I'm trying to use a VGG16 model (only predict, no training) on a CPU cluster. However, since it is a shared cluster I have to limit the number of cores used. From googling around, I've landed on the following sample code.

```
from keras.applications.vgg16 import VGG16
from keras import backend as K
import numpy as np

conf = K.tf.ConfigProto(device_count={'CPU': 1}, 
                        intra_op_parallelism_threads=2, 
                        inter_op_parallelism_threads=2)
K.set_session(K.tf.Session(config=conf))
model = VGG16(weights='imagenet', include_top=False)
x = np.random.randn(1000, 224, 224, 3)  # example data
features = model.predict(x)
```

I've tried various variations on this, including setting `allow_soft_placement=True`. However, none of it seems to work: all 128 logical cores on the cluster are used. Might be related to [this issue](https://github.com/tensorflow/tensorflow/issues/4455). 

I'm not sure if this is a keras bug, tensorflow bug or me doing something wrong. Anyway, it would be nice if there was a way to choose the number of CPUs (or GPUs) used using keras, independent of the backend.

Versions:
Tensorflow: 1.6.0
Keras: 2.1.5

**Comment by heisenbug601** [comment]
`      
        import numpy as np
	from numpy import linalg as LA
	from keras.applications.vgg16 import VGG16
	from keras.preprocessing import image
	from keras.applications.vgg16 import preprocess_input
	from keras import backend as K
	
	conf = K.tf.ConfigProto(device_count={'CPU': 2},
	                        intra_op_parallelism_threads=2,
	                        inter_op_parallelism_threads=2)
	K.set_session(K.tf.Session(config=conf))
	
	
	class VGGNet:
	    def __init__(self):
	        # weights: 'imagenet'
	        # pooling: 'max' or 'avg'
	        # input_shape: (width, height, 3), width and height should >= 48
	        self.input_shape = (224, 224, 3)
	        self.weight = 'imagenet'
	        self.pooling = 'max'
	        self.model = VGG16(weights=self.weight, input_shape=(self.input_shape[0], self.input_shape[1], self.input_shape[2]),
	                           pooling=self.pooling, include_top=False)
	        self.model.predict(np.zeros((1, 224, 224, 3)))
	
	    '''
	    Use vgg16 model to extract features
	    Output normalized feature vector
	    '''
	    def extract_feat(self, img_path):
	        img = image.load_img(img_path, target_size=(self.input_shape[0], self.input_shape[1]))
	        img = image.img_to_array(img)
	        img = np.expand_dims(img, axis=0)
	        img = preprocess_input(img)
	        feat = self.model.predict(img)
	        norm_feat = feat[0]/LA.norm(feat[0])
	        return norm_feat
	        # return feat[0]`

maybe you can try like this

**Comment by wdobbels** [comment]
I'm closing this issue as I think it is no longer relevant. I was not able to reproduce the bug on later versions (see [stackoverflow thread](https://stackoverflow.com/questions/49677913/keras-tensorflow-limit-number-of-cores-intra-op-parallelism-threads-not-wor)). It should be working now and/or the preferable API has changed since tensorflow 2.
